frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/backing-storage-step/backing-storage.tsx

```diff
@@ -0,0 +1,189 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { TFunction } from 'i18next';
+import { Form, FormSelect, FormSelectOption, FormSelectProps, Radio } from '@patternfly/react-core';
+import { StorageClassDropdown } from '@console/internal/components/utils/storage-class-dropdown';
+import { StorageClassResourceKind } from '@console/internal/module/k8s';
+import './backing-storage.scss';
+import {
+  ODF_EXTERNAL_PROVIDERS,
+  ExternalProvider,
+  StorageClusterIdentifier,
+  ExternalProviderId,
+} from '../../../../odf-external-providers/external-providers';
+import { StorageSystemKind } from '../../../../types';
+import { filterSCWithoutNoProv } from '../../../../utils/install';
+import { WizardCommonProps, WizardReducer, WizardState, WizardDispatch } from '../../reducer';
+import { BackingStorageType } from '../../../../constants/create-storage-system';
+import { ErrorHandler } from '../../error-handler';
+
+const ExternalSystemSelection: React.FC<ExternalSystemSelectionProps> = ({
+  dispatch,
+  providersList,
+  selectedProvider,
+  t,
+}) => {
+  const handleSelection: FormSelectProps['onChange'] = (value: ExternalProviderId) =>
+    dispatch({
+      type: 'backingStorage/setExternalProvider',
+      payload: value,
+    });
+
+  return (
+    <FormSelect
+      aria-label={t('ceph-storage-plugin~Select external system from list')}
+      value={selectedProvider}
+      className="odf-backing-storage__selection--width"
+      onChange={handleSelection}
+    >
+      {providersList.map((p) => (
+        <FormSelectOption key={p.id} value={p.id} label={p.label} />
+      ))}
+    </FormSelect>
+  );
+};
+
+type ExternalSystemSelectionProps = {
+  dispatch: React.Dispatch<React.ReducerAction<WizardReducer>>;
+  selectedProvider: WizardState['backingStorage']['externalProvider'];
+  providersList: ExternalProvider[];
+  t: TFunction;
+};
+
+const StorageClassSelection: React.FC<StorageClassSelectionProps> = ({ dispatch, selected }) => {
+  const onStorageClassSelect = (sc: StorageClassResourceKind) =>
+    dispatch({
+      type: 'wizard/setStorageClass',
+      payload: { name: sc?.metadata?.name, provisioner: sc?.provisioner },
```

Comment:
```suggestion
      payload: { name: sc.metadata.name, provisioner: sc.provisioner },
```
These are always defined, right? If a user selects a storage class then the object is present.

Reply:
Not really, there is an option for default storage class which is an empty object.

Reply:
I think in that case as well the sc.provisioner will be added after figuring out the default storage class from the backend.

Reply:
I meant no storage class, updated!

Reply:
Y'all can check @vbnrh 's comment, he posted the error as well.